src/codegen/properties/attribute.py

```diff
@@ -221,61 +223,76 @@ def read_buffer_size(self):
         """
-        The python data_type of the attribute.
+        str: The python data_type of the attribute.
+        
         Currently this is used in the generation of the doc string for the attribute.
         """
         return self._python_data_type
 
     @property
     def ctypes_data_type(self):
         """
-        The type of the attribute as per the ctypes definition in python.
+        str: The type of the attribute as per the ctypes definition in python.
+        
         This is used to provide the type of the attribute when making c function calls in python.
         """
         return self._ctypes_data_type
 
-class Parameter():
-    def __init__(self, name, parameter_metadata):
-        self._handle_name = name
-        self._accessor = parameter_metadata["accessor"]
-        self._ctypes_data_type = parameter_metadata["ctypes_data_type"]
-        self._cvi_name = parameter_metadata["cvi_name"]
+    def get_lib_importer_type(self):
+        return 'windll' if self.calling_convention=='StdCall' else 'cdll'
 
-    @property
-    def handle_name(self):
-        """
-        The key of the parameter.
-        """
-        return self._handle_name
+    def get_argument_types(self):
+        argtypes = []
+        for handle_parameter in self.handle_parameters:
+            if handle_parameter.ctypes_data_type == 'ctypes.c_char_p':
+                argtypes.append('ctypes_byte_str')
+            else:
+                argtypes.append(handle_parameter.ctypes_data_type)
 
-    @property
-    def accessor(self):
-        """
-        Defines how to access the handle parameter.
-        This value would be directly substituted when trying to use the handle parameter
-        """
-        return self._accessor
+        if (self.is_list and self.ctypes_data_type != 'ctypes.c_char_p' and
+                self.bitfield_enum is None):
+            argtypes.append("wrapped_ndpointer(dtype={0}, flags=('C','W'))"
+                            .format(self.ctypes_data_type))
+        
+        elif self.ctypes_data_type == 'ctypes.c_char_p':
+            argtypes.append(self.ctypes_data_type)
+        
+        else:
+            argtypes.append('ctypes.POINTER({0})'.format(self.ctypes_data_type))
 
-    @property
-    def ctypes_data_type(self):
-        """
-        Defines the ctypes data_type of the handle parameter.
-        This is used when mentioning the data_type of the handle parameter.
-        """
-        return self._ctypes_data_type
+        if self.has_explicit_read_buffer_size:
+            argtypes.append('ctypes.c_uint')
 
-    @property
-    def cvi_name(self):
-        """
-        The cvi name of the parameter.
-        This is kept for the gRPC client implementation.
-        """
-        return self._cvi_name
\ No newline at end of file
+        return argtypes
+    
+    def get_return_type(self):
+        if self.is_enum and not self.is_list:
+            return 'class: {0}'.format(self.enum)
```

Comment:
>  For example, the expected docstring is class: nidaqmx.system.scale.Scale: documentation

Take a closer look. It uses `:class:X` not `class: X`.

![image](https://user-images.githubusercontent.com/17914399/219741426-25326cef-b5be-444d-8d2e-fbc287d9c2db.png)

Reply:
Thanks for pointing this out. Any particular reason why we are following this format for class/enum objects alone?

Reply:
It's the format Sphinx uses for cross-references: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#role-py-class

The format is really `:py:class:` but Sphinx lets us omit `:py` because it's the default domain: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#directive-default-domain